Maven compile is failing

### Expected Behaviour
Project should compile succesfully

### Actual Behaviour
It's throwing error stating 
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project cif-connector-graphql: There are test failures.
[ERROR]
[ERROR] Please refer to C:\Users\bhavidya\Documents\commerce-cif-connector-master\bundles\cif-connector-graphql\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.

### Reproduce Scenario (including but not limited to)
Give the command mvn clean install

#### Steps to Reproduce

#### Platform and Version
Master version, windows platform

#### Sample Code that illustrates the problem

#### Logs taken while reproducing problem
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project cif-connector-graphql: There are test failures.
[ERROR]
[ERROR] Please refer to C:\Users\bhavidya\Documents\commerce-cif-connector-master\bundles\cif-connector-graphql\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.

Hi,

I've got the same issue on the Centos 7.

Apache Maven 3.6.2 (40f52333136460af0dc0d7232c0dc0bcf0d9e117; 2019-08-27T15:06:16Z)
Maven home: /opt/apache-maven-3.6.2
Java version: 1.8.0_242, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.242.b08-0.el7_7.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-957.12.2.el7.x86_64", arch: "amd64", family: "unix"

I've just tried to build the issues/cif-1336 and result is unfortunately the same.

Logs:
[ERROR] Failures: 
[ERROR]   GraphqlResourceProviderTest.testFactoryInitMethod:199 
resourceMapper.buildAllCategoryPaths();
Wanted 1 time:
-> at com.adobe.cq.commerce.graphql.resource.GraphqlResourceProviderTest.testFactoryInitMethod(GraphqlResourceProviderTest.java:199)
But was 2 times. Undesired invocation:
-> at com.adobe.cq.commerce.graphql.resource.GraphqlResourceProviderTest$2.run(GraphqlResourceProviderTest.java:179)

[INFO] 
[ERROR] Tests run: 110, Failures: 1, Errors: 0, Skipped: 0

...

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project cif-connector-graphql: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/aem/aem-projects/commerce-cif-connector/bundles/cif-connector-graphql/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M4:test (default-test) on project cif-connector-graphql: There are test failures.

Please refer to /home/aem/aem-projects/commerce-cif-connector/bundles/cif-connector-graphql/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.




Regards,
Michael

--- a/bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/graphql/core/GraphqlProductViewHandlerTest.java
+++ b/bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/graphql/core/GraphqlProductViewHandlerTest.java
@@ -50,6 +50,8 @@ import static com.adobe.cq.commerce.graphql.search.CatalogSearchSupport.PN_CATAL
 
 public class GraphqlProductViewHandlerTest {
 
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
     SlingHttpServletRequest servletRequest;
     ResourceResolver resourceResolver;
     TagManager tagManager;
@@ -182,11 +184,11 @@ public class GraphqlProductViewHandlerTest {
         // one hit with itemResourceType and resource
         hit.set("resource", Mockito.mock(Resource.class));
         output = viewHandler.generateHtmlOutput(hits, servletRequest, servletResponse);
-        Assert.assertEquals("Hit\n", output);
+        Assert.assertEquals("Hit" + LINE_SEPARATOR, output);
 
         // more hits with itemResourceType and resource
         hits.add(hit2);
         output = viewHandler.generateHtmlOutput(hits, servletRequest, servletResponse);
-        Assert.assertEquals("Hit\nHit\n", output);
+        Assert.assertEquals("Hit" + LINE_SEPARATOR + "Hit" + LINE_SEPARATOR, output);
     }
 }

